spec/requests/post_spec.rb

```diff
@@ -0,0 +1,164 @@
+require 'rails_helper'
+require_relative '../support/devise'
+
+RSpec.describe PostsController, type: :controller do
+    describe "GET /" do
+        login_user
+        let!(:user_post) { FactoryBot.create(:post) }
+
+        let!(:user_posts) { FactoryBot.create_list(:post,5) }
+
+        let!(:params) { { post: { content: "<div>Game</div>", title: "Content For Post"}}}
+
+        describe "Post Index" do
+            before do
+                get :index
+            end
+
+            context 'index' do
+
+                it 'is expected to get all post ' do
+                    expect(assigns[:posts].length).to be(6)
+                end
+        
+                it 'is expected to render index template' do
+                    is_expected.to render_template(:index)
+                end
+
+                it 'should return 200:OK' do
+                    get :index
+                    expect(response).to have_http_status(:success)
+                end
+                
+            end
+        end
+  
+        describe 'GET #new' do
+          before do
+            get :new
+          end
+
+          it 'is expected to assign post Comment as new instance variable' do
+            expect((assigns[:post])).to be_instance_of(Post)
+          end
+          
+          it 'should return 200:OK' do
+            expect(response).to have_http_status(:success)
+          end
+
+          it 'renders new template' do
+            is_expected.to render_template(:new)
+          end
+    
+          it 'renders application layout' do
+            is_expected.to render_template(:application)
+          end
+
+        end
+
+
+        describe 'POST #create' do
+
+            before do
+              post :create, params: params
+            end
+    
+            context 'when params are correct' do
+            
+              it 'is expected to create new post successfully' do
+                expect((assigns[:post])).to be_instance_of(Post)
+              end
```

Comment:
Requests specs are not written in that manner, these are written same like controller specs, in requests specs we use type of request and there is a difference in the code manner and arrangement, learn them and rewrite them.